### ISSUE TITLE
"No such keymap 'menuselect'" errors when using prezto/editor module

### ISSUE THREAD

**Issue by patrickli** [marlonrichert/zsh-snap#63]
* `zsh-autocomplete` version: `fdc21c0bdf4be5ae5ddbfa5c781ddf07b73c4d0b`
* Zsh version: `debian/5.8-6`
* Framework: Mixed. Mainly prezto loaded via znap
* Plugin manager: Znap

```zsh
$ git -C ~zsh-autocomplete switch main  # Make sure you test with the `main` branch.
$ git -C ~zsh-autocomplete pull         # Update to the latest commit.
$ cd $(mktemp -d)                       # Create a temp dir and enter it.
$ unset _comp_dumpfile ZDOTDIR XDG_CACHE_HOME XDG_CONFIG_HOME XDG_DATA_HOME
$ HOME=$PWD exec zsh -f       # Restart Zsh without config files in this dir.
$ source /path/to/znap/znap.zsh
$ znap source sorin-ionescu/prezto modules/editor
$ znap source marlonrichert/zsh-autocomplete
```

Error message right after you source this project.
```
.autocomplete.key-binding:bindkey:66: no such keymap `menuselect'
.autocomplete.key-binding:bindkey:67: no such keymap `menuselect'
.autocomplete.key-binding:bindkey:70: no such keymap `menuselect'
.autocomplete.key-binding:bindkey:71: no such keymap `menuselect'
.autocomplete.key-binding:bindkey:72: no such keymap `menuselect'
.autocomplete.key-binding:bindkey:73: no such keymap `menuselect'
.autocomplete.key-binding:bindkey:82: no such keymap `menuselect'
.autocomplete.key-binding:bindkey:83: no such keymap `menuselect'
.autocomplete.key-binding:bindkey:84: no such keymap `menuselect'
.autocomplete.key-binding:bindkey:85: no such keymap `menuselect'
.autocomplete.key-binding:bindkey:86: no such keymap `menuselect'
.autocomplete.key-binding:bindkey:87: no such keymap `menuselect'
.autocomplete.async:bindkey:13: no such keymap `menuselect'
.autocomplete.async:bindkey:14: no such keymap `menuselect'
.autocomplete.key-binding.precmd:bindkey:31: no such keymap `menuselect'
.autocomplete.key-binding.precmd:bindkey:32: no such keymap `menuselect'
```

**Comment by marlonrichert** [comment]
This is not a bug in `zsh-autocomplete`, but actually an incompatibility between Prezto and Znap. I should be able to fix it, though.

**Comment by marlonrichert** [comment]
Fixed.

**Comment by patrickli** [comment]
Thanks for this. However I feel like this is too much of a hack.

> This fix is specifically for people who use znap. If they don't then they will still encounter this problem.
> Would it be easier if you just check if the menuselect keymap exists? If not just recreate it. I'm not too familiar with how zsh works so that might be an invalid idea.

**Comment by patrickli** [comment]
I haven't tested this but prezto doesn't explicitly refer to menuselect at all anywhere in the code. So I'm not sure this fix will actually do anything.

**Comment by patrickli** [comment]
@marlonrichert I think I've made this bug report in the wrong place. This shouldn't be the job for a plugin manager.

How about we revert this and I will create an issue in zsh-autocomplete? Maybe I'll even do a PR.

**Comment by marlonrichert** [comment]
@patrickli 

> This fix is specifically for people who use znap. If they don't then they will still encounter this problem.

No, they won’t. I tested it and the bug doesn’t occur without Znap.

> I think I've made this bug report in the wrong place. This shouldn't be the job for a plugin manager.

You filed this bug report against `zsh-autocomplete`, but I transferred it to the Znap repo, because of the reason above.

> Would it be easier if you just check if the menuselect keymap exists? If not just recreate it. I'm not too familiar with how zsh works so that might be an invalid idea.

No, that wouldn’t work. Zsh doesn’t let you run a module more than once. What I’m doing instead is to run the `complist` module on demand when any code tries to modify the `menuselect` keymap. That was already the intention of the old code; I’m just doing it a bit more smartly now.

> How about we revert this and I will create an issue in zsh-autocomplete? Maybe I'll even do a PR.

The problem does not lie in `zsh-autocomplete`. The problem is that Znap calls `zmodload zsh/complist`, which (among other things) initializes the `menuselect` keymap, but then Prezto calls `bindkey -d`, which resets all keymaps to default state and deletes all non-default keymaps.

I fixed it by making it so that `complist` gets called only when any code actually tries to use the `menuselect` keymap (instead of calling it the first time `bindkey` gets called).
